app/scripts/lib/setupSentry.js

```diff
@@ -93,11 +97,53 @@ function getClientOptions() {
     // we can safely turn them off by setting the `sendClientReports` option to
     // `false`.
     sendClientReports: false,
-    tracesSampleRate: METAMASK_DEBUG ? 1.0 : 0.01,
+    tracesSampleRate: getTracesSampleRate(sentryTarget),
     transport: makeTransport,
   };
 }
 
+/**
+ * Compute the tracesSampleRate depending on testing condition.
+ *
+ * @param {string} sentryTarget
+ * @returns tracesSampleRate to setup Sentry
+ */
+function getTracesSampleRate(sentryTarget) {
+  if (sentryTarget === SENTRY_DSN_FAKE) {
+    return 1.0;
+  }
+
+  const flags = getManifestFlags();
+
+  if (flags.circleci) {
+    return 0.001;
+  }
+
+  if (METAMASK_DEBUG) {
+    return 1.0;
+  }
+
+  return 0.01;
+}
+
+/**
+ * Get CircleCI tags passed from the test environment, through manifest.json,
+ * and give them to the Sentry client.
+ */
+function setCircleCiTags() {
+  const { circleci } = getManifestFlags();
+
+  Sentry.setTag('circleci.enabled', Boolean(circleci?.enabled));
```

Comment:
Minor, could we early return if `enabled` is false and not change the tags at all?

Do we want this set as `false` on every single trace and error, could we infer no value is `false`?

Reply:
I don't know my way around Sentry filtering and queries that well, but I was sort of guessing that we might want to divide between `circleci.enabled=true` and `circleci.enabled=false`.  Is it just as easy in Sentry to divide between `circleci.enabled=true` and `circleci.enabled=unset`?

Reply:
I've confirmed we can search for specific traces in Sentry with `circleci.enabled is not True` which includes both `False` and missing.